pushover_open_client/pushover_open_client.py

```diff
@@ -220,15 +220,20 @@ def __init__(self, configFile=None, email=None, password=None):
 
             self.email = jsonConfig["email"]
             self.password = jsonConfig["password"]
+
+            if "twofa" in jsonConfig:
+                self.twofa = jsonConfig["twofa"]

```

Comment:
So, the twofa thing is indeed a bit tricky because, as you assumed correctly, the token is not valid for a very long time. Hence it is not being written back to the config but removed after the post-login-config was written to disk. It's a one-time-token anyway.

We could add `twofa` as an option to login for people who'd use this library to build something that has a UI. For me, this was not the case, hence I did not include this in the PR.

The login procedure for a UI would require the caller to:

- login without twofa
- handle the possible error arising from the lack of a twofa token
- tell the user to enter his twofa token
- do the login again with the twofa token

However, for my case it's just fine if the user adds his twofa token to the config and then (quickly :-)) runs the application to log in. After that he won't need to do it again for as long as the login-secret won't be invalidated.